src/psyclone/psyGen.py

```diff
@@ -3083,6 +3083,151 @@ def scalars(self):
             "Arguments.scalars must be implemented in sub-class")
 
 
+class DataAccess(object):
+    '''A helper class to simplify the determination of dependencies due to
+    overlapping accesses to data associated with instances of the
+    Argument class.
+
+    '''
+
+    def __init__(self, arg):
+        '''Store the argument associated with the instance of this class and
+        the Call, HaloExchange or GlobalSum (or a subclass thereof)
+        instance to which the argument is associated.
+
+        :param arg: the argument that we are concerned with. An \
+        argument can be found in a `Call` a `HaloExchange` or a \
+        `GlobalSum` (or a subclass thereof)
+        :type arg: :py:class:`psyclone.psyGen.Argument`
+
+        '''
+        # the `psyclone.psyGen.Argument` we are concerned with
+        self._arg = arg
+        # the call (Call, HaloExchange, or GlobalSum (or subclass)
+        # instance to which the argument is associated
+        self._call = arg.call
+        self.reset_coverage()
```

Comment:
Pylint isn't happy that `self._covered` (and others) are not defined in the constructor - effectively they get created in this call to `reset_coverage()`.

Reply:
I think pylint got out of bed on the wrong side. Oh well, I have added them to avoid its grump.